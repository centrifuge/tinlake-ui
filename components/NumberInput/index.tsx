import { FunctionComponent } from 'react';
import NumberFormat, { NumberFormatValues } from 'react-number-format';
import { TextInput } from 'grommet';

interface Props {
  value: string;
  precision?: number;
  prefix?: string;
  suffix?: string;
  onValueChange?: (values: NumberFormatValues) => void;
  [key: string]: any;
}

const NumberInput: FunctionComponent<Props> = ({
  value, precision, prefix, suffix, onValueChange, ...rest }: Props) => {
    console.log("moin moin",value);
  return <NumberFormat
    thousandSeparator="," decimalScale={precision} fixedDecimalScale
    allowNegative={false} prefix={prefix} suffix={suffix} customInput={TextInput}
    value={value} onValueChange={onValueChange} {...rest}
  />;
};

NumberInput.defaultProps = {
  value: '0',
  precision: 2,
  prefix: '',
  suffix: ''
};

export default NumberInput;
