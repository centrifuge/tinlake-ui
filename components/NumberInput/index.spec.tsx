import React from 'react';
import NumberInput from '.';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<NumberInput value="134144" precision={3} prefix="o" suffix=" CCC"
      onValueChange={({ value }) => console.log(value) } />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
