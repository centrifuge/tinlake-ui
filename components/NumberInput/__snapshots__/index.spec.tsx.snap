// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
>
  <input
    autoComplete="off"
    className="StyledTextInput-sc-1x30a0s-0 kvYrcH"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseUp={[Function]}
    type="text"
    value="o134,144.000 CCC"
  />
</div>
`;
